networkx/relabel.py

```diff
@@ -114,13 +116,9 @@ def relabel_nodes(G, mapping, copy=True):
     --------
     convert_node_labels_to_integers
     """
-    # you can pass a function f(old_label) -> new_label
-    # or a class e.g. str(old_label) -> new_label
-    # but we'll just make a dictionary here regardless
-    # To allow classes, we check if __getitem__ is a bound method using __self__
-    if not (
-        hasattr(mapping, "__getitem__") and hasattr(mapping.__getitem__, "__self__")
-    ):
+    # you can pass any callable e.g. f(old_label) -> new_label or
+    # e.g. str(old_label) -> new_label, but we'll just make a dictionary here regardless
+    if isinstance(mapping, Callable):
```

Comment:
More rabbit hole digging.  There is a Python builtin function named `callable` which avoids (or maybe implements) the whole "how do we know if this thing is a class or a function or a method or what?", and it means that people don't have to worry about registering their object as a subclass of `abc.Callable`.
We would still need to note the change in the API due to treatment of objects that are both callable and dict-like.
```python
if callable(mapping): 
```

Reply:
Good catch - that's even better as it removes the import from collections.abc.